inc/integrations/class-coral.php

```diff
@@ -584,4 +627,162 @@ public function delete_cached_values( $post_id, $post ) {
 	public function get_post_type() {
 		return $this->post_type;
 	}
+
+	/**
+	 * Hook into added post meta and identify additions of `_wp_old_slug`.
+	 * Pass along the updated permalink to Coral so that the post continues to
+	 * be properly identified.
+	 *
+	 * @param int    $meta_id     ID of updated metadata entry.
+	 * @param int    $object_id   ID of the object metadata is for.
+	 * @param string $meta_key    Metadata key.
+	 */
+	public function check_for_updated_permalink( $meta_id, $object_id, $meta_key ) {
+		// Verify the meta key.
+		if ( '_wp_old_slug' !== $meta_key ) {
+			return;
+		}
+
+		// Validate Coral is in use.
+		$coral_url = untrailingslashit( Integrations\get_option_value( 'coral', 'url' ) );
+		if ( empty( $coral_url ) ) {
+			return;
+		}
+
+		// Attempt to update the permalink.
+		$response = $this->update_permalink_in_coral( $object_id );
```

Comment:
The only remaining edge case that comes to mind is handling posts that aren't using the post id as the unique coral ID (basically, any post that exists before these PRs merge to production).

So what happens if the `id: {$post_id},` is not found in the graphql request? Does everything fail gracefully?

Reply:
Yeah, I tested under this scenario and it fails gracefully. The graphql request returns
```
{
  "data": {
    "updateStory": {
      "story": null
    }
  }
}
```
if the ID doesn't exist.

But what really happens here is accessing the post at the new permalink leads to the Lazy Story creation of another entry in Coral (so new comment thread). This is the same behavior as before, so I felt like it was enough of an edge case to not bother addressing. Any newly created posts after this merge will behave as we want, any preexisting posts will have preexisting behavior.

Reply:
Right, that sounds good to me.